staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -71,22 +71,53 @@ type EgressSelection struct {
 
 // Connection provides the configuration for a single egress selection client.
 type Connection struct {
-	// type is the type of connection used to connect from client to network/konnectivity server.
-	// Currently supported values are "http-connect" and "direct".
-	Type string `json:"type"`
+	// Protocol is the protocol used to connect from client to the konnectivity server.
+	Protocol ProtocolType
 
-	// httpConnect is the config needed to use http-connect to the konnectivity server.
-	// Absence when the type is "http-connect" will cause an error
-	// Presence when the type is "direct" will also cause an error
+	// TCP is the TCP configuration for communicating with the konnectivity server via TCP
+	// Field is optional but one of TCP or UDS must be present if Protocol is http-connect or GRPC
 	// +optional
-	HTTPConnect *HTTPConnectConfig `json:"httpConnect,omitempty"`
+	TCP *TCPTransport
```

Comment:
After reading the [union](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#unions) convention more closely, I think we should put all union fields into a struct:
```golang
type Connection struct {
  Protocol ProtocolType
  Transport *Transport
}

type Transport struct {
  TCP *TCPTransport
  UDS *UDSTransport
}

type TCPTransport struct {
  URL string
  TLSConfig *TLSConfig
}

type UDSTransport {
  UDSName string
}
```

Sorry for going back adn forth on this.

Reply:
Fixed. Is this version okay to proceed with api review?